### PR TITLE
[CI] check size of the wheels

### PR DESCRIPTION
**Pull request by simon-mo**
To catch #4304 early.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 #################### BASE BUILD IMAGE ####################
 # prepare basic build environment
-FROM nvidia/cuda:12.1.0-devel-ubuntu22.04 AS dev
+FROM nvidia/cuda:12.4.1-devel-ubuntu22.04 AS dev
```
Comment (simon-mo) [inline Dockerfile:6]: 12.1.0 is EOL

File: .buildkite/check-wheel-size.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+def check_wheel_size(directory):
+    for root, _, files in os.walk(directory):
+        for f in files:
+            if f.endswith(".whl"):
+                wheel_path = os.path.join(root, f)
+                wheel_size = os.path.getsize(wheel_path)
+                if wheel_size > MAX_SIZE:
+                    print(
+                        f"Wheel {wheel_path} is too large ({wheel_size} bytes) "
+                        f"compare to the allowed size ({MAX_SIZE} bytes).")
```
Comment (youkaichao) [inline .buildkite/check-wheel-size.py:15]: print the value in MB, so that it looks more human-friendly?

**Review by youkaichao — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (youkaichao) [reply Dockerfile:6]: I think cuda has backward compatibility, compiling for 12.1 should work for 12.4 during runtime. But the reverse might not hold. If we compile the wheel with cuda 12.4, not sure if it works for 12.1 during runtime.

**Comment by simon-mo** [comment]
Already catching it 🤦 

```
#34 [build 14/15] RUN python3 check-wheel-size.py dist
--
  | #34 0.175 Wheel dist/vllm-0.4.1+cu124-cp310-cp310-linux_x86_64.whl is too large (140229157 bytes) compare to the allowed size (104857600 bytes).
  | #34 ERROR: process "/bin/sh -c python3 check-wheel-size.py dist" did not complete successfully: exit code: 1
  | ------
  | > [build 14/15] RUN python3 check-wheel-size.py dist:
  | 0.175 Wheel dist/vllm-0.4.1+cu124-cp310-cp310-linux_x86_64.whl is too large (140229157 bytes) compare to the allowed size (104857600 bytes).
  | ------
```

**Pull request merged**